ledger/eval.go

```diff
@@ -1043,10 +1025,15 @@ func eval(ctx context.Context, l ledgerForEvaluator, blk bookkeeping.Block, vali
 
 	// If validating, do final block checks that depend on our new state
 	if validate {
-		close(txvalidator.txgroups)
-		err, gotErr := <-txvalidator.done
-		if gotErr && err != nil {
-			return StateDelta{}, err
+		// wait for the validation to complete.
+		select {
+		case err, open := <-txvalidator.done:
```

Comment:
no, if you'll look at the code, you'll see that it closes the channel when it's done and there is no validation error.
You could ask then, why do we need `if err != nil {`, since we already know it would be a non-nil; but I don't really think it adding that extra `if` statement would hurt much here.